# ３量子ビットの量子もつれGHZ状態
グリーンバーガー＝ホーン＝ツァイリンガー状態、GHZ状態は３量子ビットのもつれが出る状態です。早速実装をみてみましょう。

In [90]:
!pip install blueqat
from blueqat import Circuit                                                            

Circuit().h[:2].cx[1,2].cx[0,2].h[:].m[:].run(shots=100)

Counter({'000': 47, '111': 53})

実際に000と111が半々ででてきました。以上です。

# 概要

## GHZ状態
３量子ビットのもつれとしてGHZ状態というものがあります。
具体的な式は以下のようになります。

<img src="./img/011/011_02_0.png" width="23%">

式の導出は難しいですが以下のようになります。

<img src="./img/011/011_02_1.png" width="60%">